### PR TITLE
improve public API

### PR DESCRIPTION
**Pull request by larsgrefer**
If the compilation is successful, the getErrorX methods are useless, and vice versa.

### CODE REVIEW

File: src/main/java/io/bit3/jsass/adapter/NativeAdapter.java [hunk 1]
```diff
@@ -61,7 +62,7 @@ public Output compile(StringContext context, ImportStack importStack)
    * @return The given {@link Output}
    * @throws CompilationException If an error was found in the given {@link Output}
    */
-  private Output checkForError(Output output) throws CompilationException {
+  private SuccessOutput checkForError(Output output) throws CompilationException {
```
Comment (larsgrefer) [inline src/main/java/io/bit3/jsass/adapter/NativeAdapter.java]: This method is the logical key of this PR.
It takes the unclassified Output and either returns SuccessOutput, if the compilation was successful, or it throws an exception

**Comment by tristanlins** [comment]
I like the idea. But I'm not sure if it is really necessary to make a `SuccessOutput` and a `ErrorOutput`.
Removing the `Output.getError*()` methods and put the information directly into the `CompilationException` may be better. If an error occurred the `Output.getCss()` and `Output.getSourceMap()` yields an empty result.

**Comment by larsgrefer** [comment]
I've kept the Output class as is, because its fields are accessed in the JNI c-code.

If an error occurs, only the ErrorOutput part of the real Output is passed around, so you can't call `getCss()` and `getSourceMap()`

**Comment by tristanlins** [comment]
I refactored the whole logic of error handling. Now the c implementation check for an error and throw a `CompilationException` instead of returning an `Output` instance.

Could you rebuild the OSX library? thx.

**Comment by larsgrefer** [comment]
@tristanlins What is the benefit of doing this in the c code?

**Comment by tristanlins** [comment]
It's just a design decision. I don't like "useless" interfaces just to hide object data. Now the API only contains one output and one exception class. This feels "cleaner" to me.

**Comment by larsgrefer** [comment]
Yes, it's *cleaner*, but more c code makes this project less platform-dependent

**Comment by tristanlins** [comment]
JNI is JNI, no matter what platform it runs ;-)

**Pull request closed without merging**